pkg/config/creds.go

```diff
@@ -18,15 +19,29 @@ type Creds struct {
 	AppKey string
 }
 
+// CredentialManager provides the credentials from the operator configuration
+type CredentialManager struct {
+	secretBackend secrets.Decryptor
+	creds         Creds
+	credsMutex    sync.Mutex
```

Comment:
I feel like the `credsMutex` can be remove if:
* `NewCredentialManager()` is also responsible to get the secret, and store it in `creds`
* `GetCredential()` only returns `creds`.

WDYT?

Reply:
Is the issue error management and retry policy ?
If `NewCredentialManager()` is invoked only once whereas `GetCredential()` is called at several places, what will the behaviour be in case of a transient error in the secret backend ?

Reply:
Like we have in Datadog-agent and DCA. the process stop with an error?